### PR TITLE
js-movie-seat-booking task

### PR DESCRIPTION
**Pull request by madmaxWMFU**
[Demo](https://madmaxwmfu.github.io/tikets-app/)

### CODE REVIEW

File: submissions/madmaxWMFU/js-movie-booking/data.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            "seat": 7,
+            "type": "comfort"  
+        },
+        {
+            "row": 11,
+            "seat": 8,
+            "type": "comfort" 
+        }
+    ]
+}
```
Comment (A-Ostrovnyy) [inline submissions/madmaxWMFU/js-movie-booking/data.json]: Add an empty line at the end of each file.

File: submissions/madmaxWMFU/js-movie-booking/app.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+const priceCart = document.querySelector('.form__total');
+
+const ticketPrice = {
+    "ordinary": 60,
+    "special": 30,
+    "comfort": 120
+};
+
+const getData = () => {
+    return fetch('./data/data.json')
```
Comment (A-Ostrovnyy) [inline submissions/madmaxWMFU/js-movie-booking/app.js:15]: You should handle fetch response status. [Read more](https://www.tjvantoll.com/2015/09/13/fetch-and-errors/)

File: submissions/madmaxWMFU/js-movie-booking/app.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    filmPoster.src = data.poster;
+    filmTitle.innerText = `"${data.film}"`;
+    filmHall.innerText = `Зал №${data.hall}`;
+
+    const [date, time] = data.datetime.split(' ');
+    filmDate.innerText = reverseDate(date);
+    filmDate.setAttribute('datetime', date);
+
+    filmTime.innerText = time;
+    filmTime.setAttribute('datetime', time);
```
Comment (A-Ostrovnyy) [inline submissions/madmaxWMFU/js-movie-booking/app.js]: Better use template string for it.

File: submissions/madmaxWMFU/js-movie-booking/app.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+const renderCart = (ticket) => {
+    const [row, seat, price] = ticket.split('-');
+    const template = `
+        <div class="form__ticket" id="${ticket}">
+            <span class="form__ticket-row">Ряд: ${row}</span>    
+            <span class="form__ticket-seat">Місце: ${seat}</span>
+            <span class="form__ticket-price">Ціна: ${ticketPrice[price]}</span>
+        </div>`;
+    var ticketWrap = document.createElement("template");
```
Comment (A-Ostrovnyy) [inline submissions/madmaxWMFU/js-movie-booking/app.js]: Use `let` or `const` for defining variables.

File: submissions/madmaxWMFU/js-movie-booking/app.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    ticketWrap.innerHTML = template;
+    cart.appendChild(ticketWrap.content);
+}
+
+const deleteFromCart = (id) => {
+    Array.from(cart.children).forEach((item) => {
+        if(item.getAttribute('id') === id) {
+           item.remove(); 
+        }
+    })
```
Comment (A-Ostrovnyy) [inline submissions/madmaxWMFU/js-movie-booking/app.js]:
```suggestion
    cart.querySelectorAll('.form__ticket').forEach((item) => {
        if(item.getAttribute('id') === id) {
           item.remove(); 
        }
    })
```

Use `querySelectorAll` instead `Array.form` he has included forEach method.

File: submissions/madmaxWMFU/js-movie-booking/app.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+const init = async () => {
+    renderData(await getData());
+    createJSON();
+    hall.addEventListener('click', addToCart);
+    hall.addEventListener('touchstart', addToCart, {capture: true});
+}
+
+window.onload = init();
```
Comment (A-Ostrovnyy) [inline submissions/madmaxWMFU/js-movie-booking/app.js]: Avoid adding event listeners using `onevent` attribute use addEventListener for this.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!
See my comments below.

File: submissions/madmaxWMFU/js-movie-booking/app.js (hunk 3 above)
Comment (madmaxWMFU) [reply submissions/madmaxWMFU/js-movie-booking/app.js]: Here the data is written to already existing elements. Or is it better to redraw them?

File: submissions/madmaxWMFU/js-movie-booking/app.js (hunk 4 above)
Comment (A-Ostrovnyy) [reply submissions/madmaxWMFU/js-movie-booking/app.js]: Still not fixed)

File: submissions/madmaxWMFU/js-movie-booking/app.js (hunk 6 above)
Comment (A-Ostrovnyy) [reply submissions/madmaxWMFU/js-movie-booking/app.js]: Not fixed yet.

File: submissions/madmaxWMFU/js-movie-booking/app.js (hunk 5 above)
Comment (A-Ostrovnyy) [reply submissions/madmaxWMFU/js-movie-booking/app.js]: Not fixed yet

File: submissions/madmaxWMFU/js-movie-booking/app.js (hunk 3 above)
Comment (A-Ostrovnyy) [reply submissions/madmaxWMFU/js-movie-booking/app.js]: Better redraw instead of finding each element and adding classnames/attributes of other things.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work
Check all issues before pushing)

**Comment by madmaxWMFU** [comment]
Changed locally, but forgot to push 🤦‍♂️🤦‍♂️

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Comment by madmaxWMFU** [comment]
Thank's a lot!

**Pull request merged**